pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -350,6 +354,52 @@ func (gce *GCECloud) ensureExternalLoadBalancerDeleted(clusterName, clusterID st
 	return nil
 }
 
+func (gce *GCECloud) deleteHealthCheckAndFirewall(service *v1.Service, name, clusterID string, hcName string) error {
+	serviceName := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
+	lbRefStr := fmt.Sprintf("%v(%v)", name, serviceName)
+
+	// Check whether it is nodes health check, which has different name from the load-balancer.
+	isNodesHealthCheck := hcName != name
+	if isNodesHealthCheck {
+		// Lock to prevent deleting necessary nodes health check before it gets attached
+		// to target pool.
+		gce.sharedResourceLock.Lock()
+		defer gce.sharedResourceLock.Unlock()
+	}
+	glog.Infof("DeleteExternalTargetPoolAndChecks(%v): Deleting health check %v.", lbRefStr, hcName)
```

Comment:
Here and below, should say `deleteHealthCheckAndFirewall(...` instead.

Reply:
Changed.